lib/model/rofolder.go

```diff
@@ -70,3 +75,62 @@ func (f *sendOnlyFolder) String() string {
 func (f *sendOnlyFolder) PullErrors() []FileError {
 	return nil
 }
+
+// pull checks need for files that only differ by metadata (no changes on disk)
+func (f *sendOnlyFolder) pull() {
+	select {
+	case <-f.initialScanFinished:
+	default:
+		// Once the initial scan finished, a pull will be scheduled
+		return
+	}
+
+	f.model.fmut.RLock()
+	folderFiles := f.model.folderFiles[f.folderID]
+	ignores := f.model.folderIgnores[f.folderID]
+	f.model.fmut.RUnlock()
+
+	batch := make([]protocol.FileInfo, 0, maxBatchSizeFiles)
+	batchSizeBytes := 0
+
+	folderFiles.WithNeed(protocol.LocalDeviceID, func(intf db.FileIntf) bool {
+		if len(batch) == maxBatchSizeFiles || batchSizeBytes > maxBatchSizeBytes {
+			f.model.updateLocalsFromPulling(f.folderID, batch)
+			batch = batch[:0]
+			batchSizeBytes = 0
+		}
+
+		if ignores.ShouldIgnore(intf.FileName()) {
+			file := intf.(protocol.FileInfo)
+			file.Invalidate(f.model.id.Short())
```

Comment:
(We have this and `f.model.shortID` sprinkled both here and in rwfolder so not blaming you, but in both cases I think we should just get a `myID` in the constructor and remember it instead of reaching into the model... Doesn't need to be fixed in this PR, just mentioning it so it's out there and maybe someone remembers to clean it up)

Reply:
Done anyway (the `f.model.id.Short()` thing is likely entirely my product). I also took the opportunity to structure the folder members by function.